shared/src/domain-logic/opportunityFields.test.ts

```diff
@@ -0,0 +1,240 @@
+      async () => {
+        const foreignNexusBusinessWithoutLocationInNJ = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: false,
+          legalStructureId: "general-partnership",
+        });
+
+        const profileFields = getFieldsForProfile(foreignNexusBusinessWithoutLocationInNJ);
+        expect(profileFields).toContain("homeBasedBusiness");
+        expect(profileFields).not.toContain("municipality");
+        expect(profileFields).not.toContain("dateOfFormation");
+        expect(profileFields).not.toContain("existingEmployees");
+        expect(profileFields).not.toContain("ownershipTypeIds");
+      }
+    );
+
+    it(
+      "returns municipality, existing employees, home based business, and ownership type id keys when it is " +
+        "a starting business and the legal structure id is general-partnership",
+      async () => {
+        const startingBusiness = generateProfileData({
+          businessPersona: "STARTING",
+          legalStructureId: "general-partnership",
+        });
+
+        const profileFields = getFieldsForProfile(startingBusiness);
+        expect(profileFields).not.toContain("dateOfFormation");
+        expect(profileFields).toContain("municipality");
+        expect(profileFields).toContain("existingEmployees");
+        expect(profileFields).toContain("homeBasedBusiness");
+        expect(profileFields).toContain("ownershipTypeIds");
+      }
+    );
+
+    it(
+      "returns date of formation, municipality, existing employees, home based business, and ownership type id keys when it is " +
+        "a starting business and the legal structure id is limited-partnership",
+      async () => {
+        const startingBusiness = generateProfileData({
+          businessPersona: "STARTING",
+          legalStructureId: "limited-partnership",
+        });
+
+        const profileFields = getFieldsForProfile(startingBusiness);
+        expect(profileFields).toContain("dateOfFormation");
+        expect(profileFields).toContain("municipality");
+        expect(profileFields).toContain("existingEmployees");
+        expect(profileFields).toContain("homeBasedBusiness");
+        expect(profileFields).toContain("ownershipTypeIds");
+      }
+    );
+  });
+
+  describe("isEveryFundingFieldAnswered", () => {
+    it(
+      "returns false when municipality, existing employees, home based business, and ownership type id are all undefined and it is  " +
+        "a starting business",
+      async () => {
+        const startingBusiness = generateProfileData({
+          businessPersona: "STARTING",
+          legalStructureId: "limited-partnership",
+          existingEmployees: undefined,
+          homeBasedBusiness: undefined,
+          ownershipTypeIds: undefined,
+          municipality: undefined,
```

Comment:
[pebble] Let's be careful with these. You should look at `emptyProfileData` to match what is the empty/default value for these fields. For example, `ownershipTypeIds` is an array and is never undefined - it starts out empty. I'm kind of surprised the compiler allows us to give it an undefined value.

Reply:
changed to an empty array